gcloud-java-storage/src/main/java/com/google/gcloud/spi/StorageRpc.java

```diff
@@ -174,4 +175,8 @@ StorageObject copy(StorageObject source, Map<Option, ?> sourceOptions,
 
   void write(String uploadId, byte[] toWrite, int toWriteOffset, StorageObject dest,
       long destOffset, int length, boolean last) throws StorageException;
+
+  RewriteResponse rewrite(StorageObject source, Map<Option, ?> sourceOptions,
```

Comment:
One problem with this is that maxByteRewrittenPerCal must be the same value between calls.

We could change the RPC layer to deal with it with an open/continue model where open returns an opaque
value that includes all the information it needs for subsequent calls or we could just document it.